### ISSUE TITLE
None continious section issue

### ISSUE THREAD

**Issue by zmpulse** [robbievanleeuwen/section-properties#240]
**Describe the bug**
None continious section bringing strange results.

**To Reproduce**
1. copy paste below code
```
import sectionproperties
import sectionproperties.pre.library.steel_sections as steel_sections
import sectionproperties.analysis.section as cross_section
from sectionproperties.analysis.section import Section

from sectionproperties.pre.library.primitive_sections import rectangular_section

i_sec1 = steel_sections.i_section(d=500, b=200, t_f=16, t_w=10.2, r=21, n_r=12)
i_sec1.plot_geometry()
i_sec2 = rectangular_section(d=200, b=10.2)
i_sec2.plot_geometry()
analysis_geom = (i_sec1 - i_sec2.shift_section(94.9,50))
analysis_geom.plot_geometry()
analysis_mesh=analysis_geom.create_mesh(mesh_sizes=[5, 2.5])
section = Section(analysis_geom,analysis_mesh)
section.calculate_geometric_properties()
section.calculate_plastic_properties()
section.calculate_geometric_properties()
section.calculate_warping_properties()
section.plot_centroids()
stress_post = section.calculate_stress(N=1e3, Vy=3e3, Mxx=1e6)
stress_post.plot_stress_vm()

```

**Expected behaviour**
should increase stress from calculation without opening as below : 
![image](https://user-images.githubusercontent.com/24246172/185616115-5ecedff1-3305-4251-96f7-029a7cf44533.png)
Perhaps the results are exploding up 6.2640e+04 MPa.
![image](https://user-images.githubusercontent.com/24246172/185617809-5a078797-0dc9-49bf-969c-80c3dee561df.png)

I have added a last line in section.py in calculate_combine_stress to force the higher value than my criteria to be reassonnable with below formula:
        self.sig_vm=[213 if i>=213 else i  for i in self.sig_vm]
 
![image](https://user-images.githubusercontent.com/24246172/185618119-77190888-e427-4f69-ad9a-300af3ab2a2a.png)

It is a really strange behavior. The load don't seems to be transfer to the full section only a partial section near the cut seems to support everything.

**Additional context**
I have dig to solve the issue but I really need your help...

**Comment by zmpulse** [comment]
updated with compound geometry but seem stange again...

`import sectionproperties
import sectionproperties.pre.library.steel_sections as steel_sections
import sectionproperties.analysis.section as cross_section
from sectionproperties.analysis.section import Section
from sectionproperties.pre.geometry import CompoundGeometry

from sectionproperties.pre.library.primitive_sections import rectangular_section

i_sec1 = steel_sections.i_section(d=500, b=200, t_f=16, t_w=10.2, r=21, n_r=12)
i_sec1.plot_geometry()
i_sec2 = rectangular_section(d=200, b=10.2)
i_sec2.plot_geometry()
analysis_geom = (i_sec1 - i_sec2.shift_section(94.9,50))
analysis_geom.plot_geometry()
analysis_geomH = i_sec1 - rectangular_section(d=250, b=200)
analysis_geomH.plot_geometry()
analysis_geomB = i_sec1 - rectangular_section(d=450, b=200).shift_section(0,50)
analysis_geomB.plot_geometry()
analysis_geom= CompoundGeometry([analysis_geomH,analysis_geomB])
analysis_mesh=analysis_geom.create_mesh(mesh_sizes=[5,5])
section = Section(analysis_geom,analysis_mesh)
section.calculate_geometric_properties()
section.calculate_plastic_properties()
section.calculate_geometric_properties()
section.calculate_warping_properties()
section.plot_centroids()
stress_post = section.calculate_stress(N=1e3, Vy=2.75e5, Mxx=4.11e8)
stress_post.plot_stress_vm()`

**Comment by robbievanleeuwen** [comment]
Hi @zmpulse,

Your issue is that you are trying to anaylse shear flow through two unconnected sections which is not possible, by definition there can be no shear flow between unconnected sections so this problem is undefined.

As a result, the warping analysis cannot be performed - see note in the documentation [here](https://sectionproperties.readthedocs.io/en/latest/rst/analysis.html#warping-analysis).

As an aside, if you removed ``Vy`` from your stress analysis you should get valid stresses as *sectionproperties* should be able to calculate unconnected stresses from normal forces and bending moments.

**Comment by connorferster** [comment]
Hi @robbievanleeuwen,

I tried creating a stress analysis from disconnected sections to represent the chords on a truss but the `.calculate_stress()` method tells me that I _need_ to perform a warping analysis to calculate normal stresses on the chords.

Here is the code I used:

```python
# Geometry imports
from sectionproperties.pre.library import primitive_sections as primitives

# Analysis imports
from sectionproperties.analysis.section import Section

rect_1 = primitives.rectangular_section(b=38, d=140)
rect_2 = rect_1.shift_section(y_offset=600)
truss = rect_1 + rect_2
display(truss)
truss.create_mesh([500])
truss_sec = Section(truss, time_info=True)

truss_sec.calculate_geometric_properties()
truss_sec.calculate_plastic_properties()
truss_sec.plot_mesh()

truss_stress = truss_sec.calculate_stress(Mxx=38e6) # RuntimeError: Perform a geometric and warping analysis before carrying out a stress analysis.
```
Is this the expected behaviour?

**Comment by robbievanleeuwen** [comment]
@connorferster yes this is the expected behaviour.

If a shear force or torsion is specified for ``calculate_stress()``, we need results from the warping analysis to compute stresses. Bending moments and axial forces only require the geometric analysis to be performed. 

I guess a better way to do this would be to only require a warping analysis if the shear force or torsion is non-zero, what are your thoughts on this?

**Comment by connorferster** [comment]
I was thinking the same: check the input loads and perform the analysis based on the load and analyses available.

**Comment by robbievanleeuwen** [comment]
Proposed fix in #243.
